utils/utils.go

```diff
@@ -1044,3 +1046,27 @@ func ForkVersionAtEpoch(epoch uint64) *types.ForkVersion {
 		PreviousVersion: MustParseHex(Config.Chain.Config.GenesisForkVersion),
 	}
 }
+
+func LogError(infoIdentifier string, err error) *logrus.Entry {
+	var fileIdentifier, lineIdentifier, functionIdentifier string
+
+	pc, fullFilePath, line, ok := runtime.Caller(1)
+	if ok {
+		fileIdentifier = filepath.Base(fullFilePath)
+		functionIdentifier = runtime.FuncForPC(pc).Name()
+		lineIdentifier = strconv.Itoa(line)
+	} else {
+		msg := "Cannot read callstack"
+		fileIdentifier = msg
+		functionIdentifier = msg
+		lineIdentifier = msg
+	}
+
+	return logrus.WithFields(logrus.Fields{
+		"file":       fileIdentifier,
+		"line":       lineIdentifier,
+		"function":   functionIdentifier,
+		"info":       infoIdentifier,
+		"error type": fmt.Sprintf("%T", err),
+	}).WithError(err)
```

Comment:
The output is kinda strange, as normally the "info" is part of Error("..."), so there is now a big empty space instead.

Reply:
I rewrote the fields so that only the ones that give actual information depending on callstack and error are written.

"runtime.Caller" surely is something that takes longer than other calls but it was still just 7.5µs.

I moved the "info" part back into Error(...)/Fatal(...)